src/polynomial/mod.rs

```diff
@@ -256,6 +261,58 @@ impl<F: FiniteField, const D: usize> Polynomial<Monomial, F, D> {
       }),
     )
   }
+
+  /// Computes the Fast Fourier Transform (FFT) of a polynomial in the Monomial basis.
+  pub fn fft(&self) -> Polynomial<Lagrange<F>, F, D> {
```

Comment:
Here we can have a where clause
```suggestion
  pub fn fft(&self) -> Polynomial<Lagrange<F>, F, D> 
  where [(),D.is_power_of_two() as usize - 1]: 
  {
```

Give this a shot

Reply:
This might break compilation if we ever create a polynomial that isn't a power of two though.